tst/TurnConnectionFunctionalityTest.cpp

```diff
@@ -113,10 +133,132 @@ namespace com { namespace amazonaws { namespace kinesis { namespace video { name
         EXPECT_GE(pTurnConnection->allocationExpirationTime, GETTIME());
         MUTEX_UNLOCK(pTurnConnection->lock);
 
-        EXPECT_EQ(STATUS_SUCCESS, freeTurnConnection(&pTurnConnection));
-        EXPECT_EQ(STATUS_SUCCESS, freeConnectionListener(&pConnectionListener));
-        timerQueueFree(&timerQueueHandle);
-        deinitializeSignalingClient();
+        freeTestTurnConnection(&pTurnConnection);
+    }
+
+    TEST_F(TurnConnectionFunctionalityTest, turnConnectionSelfCleanUpWhenNotUsed)
+    {
+        PTurnConnection pTurnConnection = NULL;
+        BOOL turnReady = FALSE;
+        KvsIpAddress turnPeerAddr;
+
+        initializeTestTurnConnection(&pTurnConnection);
+
+        turnPeerAddr.port = (UINT16) getInt16(8080);
+        turnPeerAddr.family = KVS_IP_FAMILY_TYPE_IPV4;
+        turnPeerAddr.isPointToPoint = FALSE;
+        // random peer 10.1.1.1, we are not actually sending anything to it.
+        turnPeerAddr.address[0] = 0x0A;
+        turnPeerAddr.address[1] = 0x01;
+        turnPeerAddr.address[2] = 0x01;
+        turnPeerAddr.address[3] = 0x01;
+
+        EXPECT_EQ(STATUS_SUCCESS, turnConnectionAddPeer(pTurnConnection, &turnPeerAddr));
+        EXPECT_EQ(STATUS_SUCCESS, turnConnectionStart(pTurnConnection));
+
+        // wait until channel is created
+        while(!turnReady) {
+            MUTEX_LOCK(pTurnConnection->lock);
+            if (pTurnConnection->state == TURN_STATE_READY) {
+                turnReady = TRUE;
+            }
+            MUTEX_UNLOCK(pTurnConnection->lock);
+        }
+
+        EXPECT_TRUE(turnReady == TRUE);
+
+        THREAD_SLEEP((UINT64) (DEFAULT_TURN_START_CLEAN_UP_TIMEOUT * 1.5));
```

Comment:
Can we make the wait time shorter while still keep it correct?

Reply:
this is internal attribute and is not configurable right now.